Fix issue #11752

comargo opened the pull request:
Changelog: Bugfix: Fix issue #11752
Docs: https://github.com/conan-io/docs/pull/XXXX
 
Close https://github.com/conan-io/conan/issues/11752

comargo committed:
Add testcase for issue #11752

comargo committed:
Fix issue #11752

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/conan-io/conan?pullRequest=11754) <br/>All committers have signed the CLA.

lasote commented on conan/tools/cmake/cmake.py:
```diff
@@ -56,8 +56,11 @@ def __init__(self, conanfile):
         self._generator = configure_preset["generator"]
-        self._toolchain_file = configure_preset["toolchainFile"]
         self._cache_variables = configure_preset["cacheVariables"]
+        if "toolchainFile" in configure_preset:
+            self._toolchain_file = configure_preset["toolchainFile"]
+        else:
+            self._toolchain_file = self._cache_variables["CMAKE_TOOLCHAIN_FILE"]
```
It is not needed to manage the cache variable it will be applied later automatically when doing `arg_list.extend(['-D{}="{}"'.format(k, v) for k, v in self._cache_variables.items()])`

```suggestion
       self._toolchain_file = configure_preset.get("toolchainFile")
```

comargo replied:
> It is not needed to manage the cache variable

Yes, you right. I just didn't thought about it.  So final patch is much simplier.

lasote reviewed (changes requested):
Hi @comargo thanks for the PR, could you open your branch from the `release/1.51`? We are planning to open a `1.51.1` with more fixes. Thanks! EDIT: Feel free if you need to open a new PR, sometimes Github messes it up a lot. Also if you prefer us to do it just tell us! Thanks

comargo commented:
> could you open your branch from the `release/1.51`

Hello. Sorry I don't fully understand you, do you mean, that I should rebase my patch to release/1.51 and make PR as comparison of my code and release/1.51?

comargo committed:
Update conan/tools/cmake/cmake.py

It is not needed to manage the cache variable it will be applied later automatically

Co-authored-by: Luis Martinez <lasote@gmail.com>

comargo committed:
fix indent and position

comargo commented:
Hello, @lasote. I've made a draft PR https://github.com/conan-io/conan/pull/11759 that based on release/1.51 and squashes all changes to `cmake.py` to one (since final patch is just one-line-patch, not counting testcase :) )
Could you please check that I get you right.
